.github/workflows/publish.yml

```
@@ -72,7 +72,9 @@ jobs:
   releases:
     needs: [linter-master, linter-pull-request, tests]
     name: GoReleaser Build on All OS but Windows
-    if: startsWith(github.ref, 'refs/tags/')
+    # related to https://github.com/actions/runner/issues/2205
+    # since a success() is added by default and skipped jobs make success to fail we need this workaround
+    if: ${{ always() && startsWith(github.ref, 'refs/tags/') && (needs.linter-master.result == 'success' || needs.linter-pull-request.result =='success') }}
```

question: do we really need the surrounding `${{ .. }}`? From https://github.com/actions/runner/issues/2205#issuecomment-1534430719 it does not seem so
nitpick: you lack a space in both `if`

```suggestion
    if: ${{ always() && startsWith(github.ref, 'refs/tags/') && (needs.linter-master.result == 'success' || needs.linter-pull-request.result == 'success') }}
```

---

The `${{ .. }}` is needed in case of Yaml reserved character but indeed here it's not the case.